src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -366,8 +366,20 @@ public override long Position
 
         class AuthenticationCompleteException : Exception { }
 
-        string? GetPassword()
+        string? GetPassword(string? username)
```

Comment:
Username shouldn't be nullable (by this time we must already have an actual username, otherwise authentication isn't possible).

Reply:
see comment directly above for SASL auth

Reply:
Addressed above - in any case you have already made username non-nullable so all good.